### DISCUSSION TITLE
Conflict on multiple local environments

### DISCUSSION THREAD

**Discussion by Sho-ki** [prisma/prisma#16254] [Q&A]
Hi, I am using Prisma in a team development.
And I'd like to know the best practice to use Prisma with multiple branches.

## Problem
When I make changes to schema.ts in two or more branches, conflict occurs when migrating in the local environment.

#### Step 1. Create a model `User` in Branch A.
Then, run the command `npx prisma migrate dev`.

At this time
- Data is added to table _prisma_migrations
- Migration file is added to migrations folder
- User table is added to DB

#### Step 2. Switch the Branch from A to B, and create a model `Post` in  Branch B.
And run `npx prisma migrate dev`.

At this time, an error occurs because the data of _prisma_migrations and User table already exists created in Step 1.

To solve this error, I do this:
In Branch B, instead of running `npx prisma migrate dev`, I run `npx prisma db push`.
Then when I go back to Branch A, I run `npx prisma db push` to get it back.
But there is still a problem that I cannot make a migration file in Branch B because of the migration diff error.

It seems it is complicated, and I'd like to know what the best practice for Prisma team development.

**Comment by nurul3101** [comment]
Hello @Sho-ki 👋 
Did you have a chance to look at our [Team Development with Prisma Migrate](https://www.prisma.io/docs/guides/database/developing-with-prisma-migrate/team-development) Guide?

Let me know if this helps you?

**Reply by vladi-stevanovic** [reply]
👋 Hi @Sho-ki , did you perhaps have a chance to check Nurul's recommendation? Let us know if you still have any questions! 

Or, if you've found a solution, we'd love to know how you solved this problem. Feel free to post it in this thread and "Mark as answer" if this discussion is now solved!

**Reply by longlostnick** [reply]
@nurul3101 @vladi-stevanovic I have the same issue as @Sho-ki and this guide does not appear to help. The issue is how to handle AFTER the DB has been migrated, but BEFORE the team member's changes are merged. So in the case of local development, when you want to work on multiple in progress branches at the same time while using the same development DB.

**Reply by ArneCordes** [reply]
Same here. If I have 2 different feature branches with different migration states, I don't see how the guide helps?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
